pkg/hierarchy/manager_test.go

```diff
@@ -158,11 +166,51 @@ func TestManager(t *testing.T) {
 			wantCohorts: sets.New("cohort"),
 			wantCqEdge:  []cqEdge{{"queue1", "cohort"}},
 		},
+		"explicit cohort": {
+			operations: func(m M) {
+				m.AddCohort(newCohort("cohort"))
+			},
+			wantCohorts: sets.New("cohort"),
+		},
+		"delete explicit cohort": {
+			operations: func(m M) {
+				m.AddCohort(newCohort("cohort"))
+				m.DeleteCohort("cohort")
+			},
+		},
+		"delete explicit cohort idempotent": {
+			operations: func(m M) {
+				m.DeleteCohort("cohort")
+				m.AddCohort(newCohort("cohort"))
+				m.DeleteCohort("cohort")
+				m.DeleteCohort("cohort")
+			},
+		},
+		"explicit cohort persists after child deleted": {
+			operations: func(m M) {
+				m.AddClusterQueue(newCq("queue"))
+				m.UpdateClusterQueueEdge("queue", "cohort")
+				m.AddCohort(newCohort("cohort"))
+				m.DeleteClusterQueue("queue")
+			},
+			wantCohorts: sets.New("cohort"),
+		},
+		"explicit cohort downgraded to implicit cohort": {
+			operations: func(m M) {
+				m.AddCohort(newCohort("cohort"))
+				m.AddClusterQueue(newCq("queue"))
+				m.UpdateClusterQueueEdge("queue", "cohort")
+				m.DeleteCohort("cohort")
+			},
+			wantCohorts: sets.New("cohort"),
+			wantCqs:     sets.New("queue"),
+			wantCqEdge:  []cqEdge{{"queue", "cohort"}},
+		},
```

Comment:
What about testing the `cleanupCohort` method? Like: 
```go
m.AddCohort(newCohort("cohort"))
m.AddClusterQueue(newCq("queue"))
m.UpdateClusterQueueEdge("queue", "cohort")
m.DeleteCohort("cohort")
m.DeleteClusterQueue("queue")
```

Reply:
added